app.rb

```diff
@@ -0,0 +1,123 @@
+      end
+      option input
+    end
+  end
+
+  def list_all_persons
+    puts 'People List is empty.' if @persons.empty?
+    @persons.each do |person|
+      puts "Name: #{person.name}, Age: #{person.age} ID: #{person.id}"
+    end
+  end
+
+  def list_all_books
+    puts 'Books List is empty.' if @books.empty?
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def add_person
+    print 'Do you want to create a student (1) or a teacher (2) ? [Input The Number]: '
+    input = gets.chomp.to_i
+    if input == 1
+      add_student
+    elsif input == 2
+      add_teacher
+    else
+      puts 'Invalid option. Please try again'
+    end
+  end
+
+  def add_student
+    print 'Age : '
+    age = gets.chomp.to_i
+    print 'Name : '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N] : '
+    parent_permission = gets.chomp.downcase
+    print 'Classroom : '
+    classroom_lable = gets.chomp
+    classroom = Classroom.new(classroom_lable)
+    case parent_permission
+    when 'y'
+      student = Student.new(name, age, classroom, parent_permission: true)
+      @persons << student
+    when 'n'
+      student = Student.new(name, age, classroom, parent_permission: false)
+      @persons << student
+    end
+  end
+
+  def add_teacher
+    puts 'Creating a teacher'
+    print 'Age : '
+    age = gets.chomp.to_i
+    print 'Name : '
+    name = gets.chomp
+    print 'Specialization : '
+    specialization = gets.chomp
+    teacher = Teacher.new(name, age, specialization)
+    @persons << teacher
+  end
+
+  def add_book
+    puts 'Creating a book'
+    print 'Title : '
+    title = gets.chomp
+    print 'Author : '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+  end
+
+  def add_rentals
+    puts 'Select which book you want to rent by its Number '
+    @books.each_with_index { |book, index| puts "Number: #{index + 1}. Title: #{book.title}, Author: #{book.author}" }
+    book_id = gets.chomp.to_i - 1
+    puts 'Select which person will rent the book by its Number '
+    @persons.each_with_index { |person, index| puts "Number: #{index + 1}. Name: #{person.name}, Age: #{person.age}" }
+    person_id = gets.chomp.to_i - 1
+    print 'Date (YYYY/MM/DD): '
+    date = gets.chomp.to_s
+    rental = Rental.new(date, @books[book_id], @persons[person_id])
+    @rentals << rental
+    puts 'Rental created successfully'
+  end
+
+  def list_all_rentals
+    puts 'List is empty. Please add a rental' if @rentals.empty?
+    puts 'Enter the ID of the rental you want to see'
+    @persons.each { |person| puts "ID: #{person.id}, Name: #{person.name}" }
+    id = gets.chomp.to_i
+    puts 'Rented books:'
+    @rentals.each do |rental|
+      if rental.person.id == id
+        print "Person: #{rental.person.name} Date: #{rental.date} "
+        puts "Book: #{rental.book.title}, Author: #{rental.book.author}"
+      else
+        puts 'No rentals found'
+      end
```

Comment:
- [ ] Absolutely! 🚀 First of all, I appreciate your work on the rental listing feature, and it's clear that you're on the right track. However, there's a minor tweak needed to enhance the user experience and align with the requirements.

   - Currently, your application displays all available users even before prompting for the `ID` of the person whose rental they want to see. To align with the expected behavior, I suggest modifying this flow slightly. After the user selects the `option` to view rentals the first thing you should as them is the `ID` of the person whose rental they want to see without displaying all the people available.

   - For more clarity on these adjustments, I highly recommend watching the video provided in the project request. It will provide a clearer picture of how the application is intended to function and help you ensure your implementation matches the desired workflow.
   
<table>
<tr>
<th align=“middle”>Main design</th>
<th align=“middle”>Your design</th>
</tr>
<tr>
<td>
<img width="567" alt="Screenshot 2023-08-31 at 20 52 40" src="https://github.com/demesameneshoa/OOP-Library/assets/107192076/5768bc19-1b3d-4a89-911e-d80820bc5352">
</td>
<td><img width="567" alt="Screenshot 2023-08-31 at 21 37 17" src="https://github.com/demesameneshoa/OOP-Library/assets/107192076/08b97055-b3b7-4944-85f5-a09fb9274317"></td>
</tr>
</table>


Happy coding! 😊🚀🍀